example/components/ConnectWithSelect.tsx

```diff
@@ -39,152 +42,102 @@ function ChainSelect({
-          .catch(setError)
-      } else {
-        connector
-          .activate(desiredChainId === -1 ? undefined : getAddChainParameters(desiredChainId))
-          .then(() => setError(undefined))
-          .catch(setError)
+        setError(undefined)
+      } catch (error) {
+        setError(error)
       }
     },
-    [connector, chainId, setError]
+    [connector, activeChainId, setError]
   )
 
-  const onClick = useCallback((): void => {
-    setError(undefined)
-    if (connector instanceof GnosisSafe) {
-      connector
-        .activate()
-        .then(() => setError(undefined))
-        .catch(setError)
-    } else if (connector instanceof WalletConnect || connector instanceof Network) {
-      connector
-        .activate(desiredChainId === -1 ? undefined : desiredChainId)
-        .then(() => setError(undefined))
-        .catch(setError)
-    } else {
-      connector
-        .activate(desiredChainId === -1 ? undefined : getAddChainParameters(desiredChainId))
-        .then(() => setError(undefined))
-        .catch(setError)
-    }
-  }, [connector, desiredChainId, setError])
-
-  if (error) {
-    return (
-      <div style={{ display: 'flex', flexDirection: 'column' }}>
-        {!(connector instanceof GnosisSafe) && (
-          <ChainSelect
-            chainId={desiredChainId}
-            switchChain={switchChain}
-            displayDefault={displayDefault}
-            chainIds={chainIds}
-          />
-        )}
-        <div style={{ marginBottom: '1rem' }} />
-        <button onClick={onClick}>Try Again?</button>
-      </div>
-    )
-  } else if (isActive) {
-    return (
-      <div style={{ display: 'flex', flexDirection: 'column' }}>
-        {!(connector instanceof GnosisSafe) && (
-          <ChainSelect
-            chainId={desiredChainId === -1 ? -1 : chainId}
-            switchChain={switchChain}
-            displayDefault={displayDefault}
-            chainIds={chainIds}
-          />
-        )}
-        <div style={{ marginBottom: '1rem' }} />
-        <button
-          onClick={() => {
-            if (connector?.deactivate) {
-              void connector.deactivate()
-            } else {
-              void connector.resetState()
-            }
-          }}
-        >
-          Disconnect
-        </button>
-      </div>
-    )
-  } else {
-    return (
-      <div style={{ display: 'flex', flexDirection: 'column' }}>
-        {!(connector instanceof GnosisSafe) && (
-          <ChainSelect
-            chainId={desiredChainId}
-            switchChain={isActivating ? undefined : switchChain}
-            displayDefault={displayDefault}
-            chainIds={chainIds}
-          />
-        )}
-        <div style={{ marginBottom: '1rem' }} />
+  return (
+    <div style={{ display: 'flex', flexDirection: 'column' }}>
+      {!(connector instanceof GnosisSafe) && (
+        <ChainSelect activeChainId={desiredChainId} switchChain={switchChain} chainIds={chainIds} />
```

Comment:
In this case, isn't `activeChainId` not necessarily active yet, just `desired` because it may be in the process of being switched to?

Reply:
Yes, that is correct. We have a managed `<select />` component so we want to update the state optimistically to make sure select input reflects new choice. Since we also show "Try again?" button on error, it is okay to do it this way. Perhaps the prop name is a bit confusing here, we mean "activeElement" where "element" is "chainId" in this context, not "activeChainId" in web3-react terminology.